packages/compass-saved-aggregations-queries/src/stores/index.ts

```diff
@@ -1,21 +1,52 @@
 import type AppRegistry from 'hadron-app-registry';
-import { createStore, applyMiddleware, combineReducers } from 'redux';
+import {
+  createStore,
+  applyMiddleware,
+  combineReducers,
+  Store,
+  AnyAction,
+} from 'redux';
 import thunk from 'redux-thunk';
 import itemsReducer from './aggregations-queries-items';
-import instanceReducer, { resetInstance, setInstance } from './instance';
+import instanceReducer, { setInstance, resetInstance } from './instance';
+import dataServiceReducer, {
+  setDataService,
+  resetDataService,
+} from './data-service';
+import openItemReducer from './open-item';
 
 const _store = createStore(
   combineReducers({
     savedItems: itemsReducer,
     instance: instanceReducer,
+    dataService: dataServiceReducer,
+    openItem: openItemReducer,
   }),
   applyMiddleware(thunk)
 );
 
-export type RootState = ReturnType<typeof _store.getState>;
+type StoreActions<T> = T extends Store<unknown, infer A> ? A : never;
+
+type StoreState<T> = T extends Store<infer S, AnyAction> ? S : never;
+
+export type RootActions = StoreActions<typeof _store>;
+
+export type RootState = StoreState<typeof _store>;
 
 const store = Object.assign(_store, {
   onActivated(appRegistry: AppRegistry) {
+    appRegistry.on('data-service-connected', (err, dataService) => {
```

Comment:
random unrelated thought: unless we already did it would be nice to have types for these events in a central place (maybe just convert AppRegistry to ts).

Reply:
Yeah, good point. I think app-registry is already ts, but the event types are generic. I'll look into how much effort it is and might add types for a few that are used everywhere and will be low effort to type right away